lib/airbrake/rack/instrumentable.rb

```diff
@@ -14,15 +16,62 @@ module Rack
     # @api public
     # @since v9.2.0
     module Instrumentable
+      # rubocop:disable Metrics/AbcSize
+      # rubocop:disable Metrics/MethodLength
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/PerceivedComplexity
       def airbrake_capture_timing(method_name, label: method_name.to_s)
-        alias_method "#{method_name}_without_airbrake", method_name
+        # Check the visibility.
+        is_public = private_method_defined?(method_name)
+        is_protected = protected_method_defined?(method_name)
+        is_private = private_method_defined?(method_name)
+
+        # Make sure to forward keyword arguments too when using 2.7+ syntax.
+        use_kw_args = Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.7")
+        args = use_kw_args ? "*args, **kw_args, &block" : "*args, &block"
 
-        define_method(method_name) do |*args|
-          Airbrake::Rack.capture_timing(label) do
-            __send__("#{method_name}_without_airbrake", *args)
-          end
+        # Don't chain already-prepended or operator methods.
+        self_class_idx = ancestors.index(self)
+        method_owner_idx = ancestors.index(instance_method(method_name).owner)
+        if method_owner_idx < self_class_idx || /\A\W/ =~ method_name
+          prepend(
+            Module.new do
```

Comment:
While this trick is useful, it has issues. I ran the test suite and inspected ancestors of the test class that you defined. As you can see, we prepend a module every time we call `airbrake_capture_timing`. This is expected but makes debugging harder now.

```
> klass.ancestors
=> [#<Module:0x00007f8875a011a8>,
 #<Module:0x00007f8875a03750>,
 #<Module:0x00007f8875bf8948>,
 #<Module:0x00007f8875bf8ec0>,
 #<Module:0x00007f8875bf90a0>,
 #<Module:0x00007f8875bfa900>,
 #<Class:0x00007f88751d51a0>,
 Object,
 JSON::Ext::Generator::GeneratorMethods::Object,
 PP::ObjectMixin,
 Kernel,
 BasicObject]
```

I think for this PR anonymous modules will do. In the future (prior to releasing the library), we would need to solve this to improve debugging. We would probably need to wrap `Module` in our own class, so that we could control naming. Another idea is to prepend only to one module and if we are prepending again to the same class, we would actually define a method on our already prepended module, so that we don't waste memory by spawning modules with 1 method only. Either way, let's not do anything clever in this PR.

Reply:
I agree with your assessment of these issues. I've had to apply this code pattern in other projects, some of which I'm working on right now, and this is consistent with what I've seen before. It would be worth it to me to work on a better solution, as I might want to apply that solution elsewhere. Your suggestions are appreciated.

Nonetheless, I also agree that there's no need to have Ruby 2.7 compatibility wait for this. Best to get a working solution out now, and clean it up later. If I can't get a better solution out in the next round of fixes, good, but if not it should be OK.